core/src/banking_stage.rs

```diff
@@ -1339,6 +1342,7 @@ impl BankingStage {
             retryable_transaction_indexes,
             commit_transactions_result: Ok(()),
             execute_and_commit_timings,
+            execution_results,
```

Comment:
Hmm, it looks like there is still a problem when `record_transactions()` errors. It can potentially mark all tx as retryable, even the ones that executed successfully.

Reply:
So record only fails when MaxHeightReached. So those executed transactions will have to be retried in _next_slot. It should no longer impact _this_ block, does it?

Reply:
Just to clarify: `record_transactions()` tries to send a list of executed transactions to `recorder` in all-or-none fashion. And it can only fail with error `MaxHeightReached` (or panic, which is not the case we are trying to handle here). When that happens it marks all executed-yet-not-recorded transactions as retryable, but `execution_records` are not altered. Therefore we would have transactions that are executed (eg have counted units in _this_ bank's cost_tracker), but not actually recorded. To make the metrics accurate, we should remove those executed-but-not-recorded transactions' units from tracker too.

Reply:
[commit](15443cf3ebc7ba43624cfa2509dc14884bfdf36b)